### PR TITLE
Adds testgen validator set

### PR DESCRIPTION
**Pull request by Shivani912**
This PR is step 1 of #665
* [X] Referenced an issue explaining the need for the change
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [X] Wrote tests
* [ ] Updated CHANGELOG.md

### CODE REVIEW

File: testgen/src/validator_set.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+use simple_error::*;
+use tendermint::validator;
+
+#[derive(Debug, Options, Serialize, Deserialize, Clone)]
+pub struct ValidatorSet {
+    #[options(
+        parse(try_from_str = "parse_as::<Vec<Validator>>"),
+        help = "validators (required)"
+    )]
+    validators: Option<Vec<Validator>>,
```
Comment (melekes) [inline testgen/src/validator_set.rs:17]: could a validator set ever be empty?

Comment (Shivani912) [reply testgen/src/validator_set.rs:17]: Ideally, not. But in testgen we want to allow such situations to be able to test them.

Comment (andrey-kuprianov) [reply testgen/src/validator_set.rs:17]: yes, as this is primarily for testing, we do want to test situations when the validator set coming from the wire is empty.

**Review by andrey-kuprianov — CHANGES_REQUESTED** [review]
Thanks for implementing this, @Shivani912! 
As the first draft it looks ok, but it is not at all integrated with the rest of the datastructures. Here are some points:

* everywhere where `Vec<Validator>` is used, e.g. in the `Header`, `ValidatorSet` should be used instead.
 * `generate_validators` should disappear from `validator`, and become just part of `ValidatorSet::generate()`
 * `ValidatorSet` should have the options to set `total_voting_power` and `proposer`. If they are not set, they should be calculated automatically.

These are the main points for now. There are two options on how to proceed:
 1. Either you can make these changes now;
 2. Or create a follow-up issue for that (or comment in the original issue #665), and now focus on writing the unit tests for the LightClient.

**Review by andrey-kuprianov — APPROVED** [review]
After a discussion, we've decided to merge this PR now, and address the comments above at a later time point. @Shivani912 could you please put those into #665? Thanks!

**Pull request merged**